tools/ci/tests/test_pr_preview.py

```diff
@@ -81,7 +82,10 @@ def do_all(self):
         self.server.actual_traffic.append((request, response))
         self.send_response(response[0])
         self.end_headers()
-        self.wfile.write(json.dumps(response[1]).encode('utf-8'))
+        if self.server.reponse_body_is_json:
+            self.wfile.write(json.dumps(response[1]).encode('utf-8'))
+        else:
+            self.wfile.write(response[1].encode('utf-8'))
```

Comment:
Thank you for fixing this. I think the original implementation abused the fact that an int would be JSON encoded into a number literal, which is a valid git SHA...

Reply:
Yep, it was quite a fun one to find and to fix!

And by fun I mean a pain :D.